### ISSUE TITLE
No error when accessing setter-only properties

### ISSUE THREAD

**Issue by MazeChaZer** [microsoft/TypeScript#58112]
### 🔎 Search Terms

"setter only", "setter-only"

### 🕗 Version & Regression Information

- This behavior is present in the current version of TypeScript, 5.4.4

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQgUwC7QCMwAnACiIC8AuLXYgSwDsBzAShWgF9VvVh5GEfADNE0ALzRG2AO5xEpVgG4+AodABuYEAFdsNIQxYToo+ADoixJdAD0t6ABUAFvRiQI9ZowC22RvgQzvA6IAAmUvD4BNjQ2iDwMthhADSmiJYk0DL0ICBxIDJgAJ4wMdA6jGHYwkzJAIRAA

### 💻 Code

```ts
class Foo {
    set bar(baz: string) { }
}

const foo = new Foo();
const value: string = foo.bar; // This assignment should not be allowed, foo.bar will always be undefined!
```

### 🙁 Actual behavior

There is no error reported during type-checking.

### 🙂 Expected behavior

The type-checking should report an error for the assignment in the last line as `foo.bar` will always be undefined.

**Comment by MartinJohns** [comment]
Duplicate of #30852, which is marked a duplicate of #21759. See this comment: https://github.com/microsoft/TypeScript/issues/30852#issuecomment-482315524

**Comment by DanielRosenwasser** [comment]
Things have changed since those comments. `set` is encoded into the declaration files (though with some bugs - [playground link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwDYEMDOa4GE4G8BQcRcaw8IAFAG4pICuwAXCTFAJYB2A5gJR4C++QflCRYcTjGBQAZigTA4ASTyFipctVoNmaVp14BuIfhHho8JGTgQARgCs4AXlXES1yjXpMW7bnwI3QX4gA)). We should reconsider this.
